design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,309 @@
+}
+
+type CompanyUsers @doc(description: "Output data schema for an object returned by a Company users search query.") {
+    items: [CompanyUser] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "Pagination meta data.")
+}
+
+type CompanyUser @doc(description: "Company User (Customer assigned to Company) entity output data schema.") {
+    id: ID! @doc(description: "Company User/Customer id.")
+    email: String! @doc(description: "Company User email address.")
+    firstname: String! @doc(description: "Company User first name.")
+    lastname: String! @doc(description: "Company User last name.")
+    status: CompanyUserStatusEnum! @doc(description: "Company User status.")
+    job_title: String! @doc(description: "Company User job title.")
+    telephone: String! @doc(description: "Company User phone number.")
+    role: CompanyRole! @doc(description: "Company User role data (includes permissions).")
+    team: CompanyTeam @doc(description: "Company User team data.")
+}
+
+type CompanyRoles @doc(description: "Output data schema for an object returned by a Company roles search query.") {
+    items: [CompanyRole] @doc(description: "A list of company roles that match the specified search criteria.")
+    total_count: Int @doc(description: "The total number of objects matching the specified filter.")
+    page_info: SearchResultPageInfo @doc(description: "Pagination meta data.")
+}
+
+type CompanyRole @doc(description: "Company role output data schema returned in response to a query by Role id.") {
+    id: ID! @doc(description: "Role id.")
+    name: String! @doc(description: "Role name.")
+    users_count: Int @doc(description: "Total number of Users with such Role within Company Hierarchy.")
+    permissions: [String] @doc(description: "A list of permission resources defined for a Role.")
+}
+
+type CompanyAclResource @doc(description: "Output data schema for an object with Role permission resource information.") {
+    id: ID! @doc(description: "ACL resource id.")
+    text: String! @doc(description: "ACL resource label.")
+    sortOrder: Int! @doc(description: "ACL resource sort order.")
+    children: [CompanyAclResource!] @doc(description: "An array of sub-resources.")
+}
+
+type CompanyUserEmailCheckResponse @doc(description: "Response object schema for a Company User email validation query.") {
+    status: Boolean! @doc(description: "True - email is valid, False - email is not valid")
+}
+
+type CompanyHierarchyOutput @doc(description: "Response object schema for a Company Hierarchy query.") {
+    structure: CompanyHierarchyElement @doc(description: "An array of Company structure elements.")
+    draggable: Boolean @doc(description: "Flag that defines whether Company Hierarchy can be changed by current User or not.")
```

Comment:
`draggable` seems too specific to a kind of UI interaction. Should change the name to better match what the data is, rather than how the data is changed.

Reply:
fixed